crypto/crypto.go

```diff
@@ -20,8 +20,30 @@ func GetPubKeyHex(privateKey PrivateKey) string {
 	return hex.EncodeToString(privKey.PublicKey.X) + hex.EncodeToString(privKey.PublicKey.Y)
 }
 
-func GeneratePrivateKey() (PrivateKey, error) {
-	return elliptic.GeneratePrivateKey(elliptic.Secp256k1)
+func GeneratePrivateKey(difficulty int) (PrivateKey, error) {
+	for true {
+		privKey, err := elliptic.GeneratePrivateKey(elliptic.Secp256k1)
+		pubKey := sha256.Sum256(append(privKey.PublicKey.X, privKey.PublicKey.X...))
```

Comment:
How about `privKey.PublicKey.X, privKey.PublicKey.Y...`? :)
Also this is incorrect
`append(a, b...)` may modify `a` which isn't desired here. You need to create an extra byte array and copy X,Y there

Reply:
Done